packages/web3-eth-contract/src/contract.ts

```diff
@@ -1115,7 +1113,15 @@ export class Contract<Abi extends ContractAbi>
 			contractOptions: modifiedContractOptions,
 		});
 
-		return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT);
+		try {
+			return sendTransaction(this, tx, DEFAULT_RETURN_FORMAT);
```

Comment:
I believe this catch will never be triggered since we return a promise (Web3PromiEvent) and the actual logic will run when the user executes the promise. That's why I concluded using a listener in #5659 PR.

Reply:
Thanks @nikoulai for catching this deadly mistake.